### PR TITLE
#5337: Minor fix to remaining Mixtral tests due to merge conflict

### PR DESCRIPTION
**Pull request by mtairum**
Issue: https://github.com/tenstorrent/tt-metal/issues/12264

Waiting for the pipelines to complete with green status.
Will merge after.

- [x] [T3k perf](https://github.com/tenstorrent/tt-metal/actions/runs/10736842261/job/29777540699)
- [ ] [T3k perplexity](https://github.com/tenstorrent/tt-metal/actions/runs/10735927647)
- [x] [T3k frequent](https://github.com/tenstorrent/tt-metal/actions/runs/10721200837)

Unit tests and demo tests are already good.

### CODE REVIEW

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py [hunk 1]
```diff
@@ -59,8 +59,14 @@ def test_mixtral_model_perf(
 
     dtype = ttnn.bfloat8_b
 
+    batch_size = 32
+    # Although in decode-only mode change the max seqlen to 16k to avoid KV cache running out of memory size
+    max_seqlen = 16384
```
Comment (uaydonat) [inline models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py:64]: should not max_seq_len be a function of the batch size?

File: models/demos/t3000/mixtral8x7b/tt/model_config.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class TtModelArgs:
     norm_eps = 1e-05
     vocab_size = 32000
 
-    max_batch_size = 32  # default
+    max_batch_size = 8  # default
```
Comment (uaydonat) [inline models/demos/t3000/mixtral8x7b/tt/model_config.py:24]: why not 32?

File: models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py (hunk 1 above)
Comment (mtairum) [reply models/demos/t3000/mixtral8x7b/tests/test_mixtral_perf.py:64]: Yes. However this test is decode only, and for that one we are only running batch == 32.

For prefill test, we do batch_size in function of max seqlen.

File: models/demos/t3000/mixtral8x7b/tt/model_config.py (hunk 2 above)
Comment (mtairum) [reply models/demos/t3000/mixtral8x7b/tt/model_config.py:24]: max_batch_size here is defined based on the max length of 32k. This ensures that if the user tries to use the model with a new script it won't go above the limitations.

But I'll add an update with a comment on the supported batch sizes <-> seqlens to the model config, so end users are aware. 👍

**Comment by mtairum** [comment]
Only the perplexity test is failing now:

```
libc++abi: terminating due to uncaught exception of type std::runtime_error: TT_FATAL @ ../ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:972: a_shape[i] == b_shape[i] && "bmm (non-bcast matmul) expects input tensors of shapes BCMK*BCKN=BCMN or equivalent"
```

It seems to be using old code. Will have to compare it to the demo/test_model code and update it.

File: models/demos/t3000/mixtral8x7b/tt/model_config.py (hunk 2 above)
Comment (uaydonat) [reply models/demos/t3000/mixtral8x7b/tt/model_config.py]: thank you, it is clearer now.

**Review by uaydonat — APPROVED** [review]

**Pull request merged**